Replace links with buttons in pagination component

offbeatjs opened the pull request:
Updated pagination component to use buttons for navigation, enhancing user interaction with loading states.
- To prevent the loading API spam

offbeatjs committed:
Replace links with buttons in pagination component

Updated pagination component to use buttons for navigation, enhancing user interaction with loading states.
- To prevent the loading API spam

max-programming commented on src/app/(public)/repos/_components/pagination.tsx:
```diff
@@ -15,24 +18,69 @@ export function Pagination({
+  const [isLoading, setIsLoading] = useState(false);
+  const router = useRouter();
+
+  const handlePrevPage = (e: React.MouseEvent<HTMLButtonElement>) => {
```
I prefer using named functions instead of arrow functions when defining them

```suggestion
  function handlePrevPage(e: React.MouseEvent<HTMLButtonElement>) {
```

max-programming commented on src/app/(public)/repos/_components/pagination.tsx:
```diff
@@ -15,24 +18,69 @@ export function Pagination({
   totalCount,
   searchParams
 }: PaginationProps) {
+  const [isLoading, setIsLoading] = useState(false);
```
Maybe let's make use of the [`useTransition` hook](https://react.dev/reference/react/useTransition) here and wrap `router.push` calls in `startTransition()`

offbeatjs committed:
Refactor pagination component to use useTransition

refactor `pagination.tsx` by:
- adding a cleaner delta logic suggested by @max-programming 
- making sure the state resets after page is loaded

offbeatjs committed:
Merge branch 'max-programming:main' into patch-1

offbeatjs commented:
Tested, it works fine ! You can merge it
